### DISCUSSION TITLE
Enrollment via webauthn and authentication via libfido2

### DISCUSSION THREAD

**Discussion by shimritd** [Yubico/libfido2#713] [Q&A]
hi, 
i'm using HID fide NFC cards, when i'm enrolling the card via the lib (with cred) and then authenticating (with assert), everything goes well. 
But when i'm enrolling the card via the web (webauthn) and try to authenticate via the lib, i'm getting an error.
This is the log, any idea what's the problem?

Thanks! 

fido_tx: dev=000001A255D6CFE0, cmd=0x06
fido_tx: buf=000001A255D6CFE0, len=8
0000: f8 c1 e4 ae 82 ce 72 bc
fido_pcsc_write: writing: buf=000000700BAFF130, len=14
0000: 00 a4 04 00 08 a0 00 00 06 47 2f 00 01 00
fido_pcsc_write: read: buf=000001A255D8D668, len=8
0000: 55 32 46 5f 56 32 90 00
fido_rx: dev=000001A255D6CFE0, cmd=0x06, ms=-1
fido_pcsc_read: reading: buf=000001A255D8D668, len=8
0000: 55 32 46 5f 56 32 90 00
fido_dev_get_cbor_info_tx: dev=000001A255D6CFE0
fido_tx: dev=000001A255D6CFE0, cmd=0x10
fido_tx: buf=000000700BAFF404, len=1
0000: 04
fido_pcsc_write: writing: buf=000000700BAFF060, len=7
0000: 80 10 00 00 01 04 00
fido_pcsc_write: read: buf=000001A255D8D668, len=74
0000: 00 a5 01 82 66 55 32 46 5f 56 32 68 46 49 44 4f
0016: 5f 32 5f 30 02 81 6b 68 6d 61 63 2d 73 65 63 72
0032: 65 74 03 50 ae b6 56 9c f8 fb 49 50 ac 60 24 ca
0048: 2b be 2e 52 04 a2 62 72 6b f5 69 63 6c 69 65 6e
0064: 74 50 69 6e f5 06 81 01 90 00
fido_dev_get_cbor_info_rx: dev=000001A255D6CFE0, ci=000001A255D8ED00, ms=-1
fido_rx: dev=000001A255D6CFE0, cmd=0x10, ms=-1
fido_pcsc_read: reading: buf=000001A255D8D668, len=74
0000: 00 a5 01 82 66 55 32 46 5f 56 32 68 46 49 44 4f
0016: 5f 32 5f 30 02 81 6b 68 6d 61 63 2d 73 65 63 72
0032: 65 74 03 50 ae b6 56 9c f8 fb 49 50 ac 60 24 ca
0048: 2b be 2e 52 04 a2 62 72 6b f5 69 63 6c 69 65 6e
0064: 74 50 69 6e f5 06 81 01 90 00
fido_dev_open_rx: FIDO_MAXMSG=2048, maxmsgsiz=0
fido_dev_authkey_tx: dev=000001A255D6CFE0
fido_tx: dev=000001A255D6CFE0, cmd=0x10
fido_tx: buf=000001A255D770A0, len=6
0000: 06 a2 01 01 02 02
fido_pcsc_write: writing: buf=000000700BAFEFD0, len=12
0000: 80 10 00 00 06 06 a2 01 01 02 02 00
fido_pcsc_write: read: buf=000001A255D8D668, len=83
0000: 00 a1 01 a5 01 02 03 38 18 20 01 21 58 20 ce e8
0016: 38 18 31 4a 40 c3 d2 9b 71 0e 7a f8 c7 cb 53 37
0032: 90 e2 d9 dc 58 f1 ab 9c 4f 8f bf 03 52 3c 22 58
0048: 20 4d dc 2c 9f db ad 32 5a 7a 19 19 be bd c2 fc
0064: 52 c7 03 af 89 c8 b9 50 e5 43 f0 c6 79 2f 0a 88
0080: 4f 90 00
fido_dev_authkey_rx: dev=000001A255D6CFE0, authkey=000001A255D7E260, ms=-1
fido_rx: dev=000001A255D6CFE0, cmd=0x10, ms=-1
fido_pcsc_read: reading: buf=000001A255D8D668, len=83
0000: 00 a1 01 a5 01 02 03 38 18 20 01 21 58 20 ce e8
0016: 38 18 31 4a 40 c3 d2 9b 71 0e 7a f8 c7 cb 53 37
0032: 90 e2 d9 dc 58 f1 ab 9c 4f 8f bf 03 52 3c 22 58
0048: 20 4d dc 2c 9f db ad 32 5a 7a 19 19 be bd c2 fc
0064: 52 c7 03 af 89 c8 b9 50 e5 43 f0 c6 79 2f 0a 88
0080: 4f 90 00
fido_tx: dev=000001A255D6CFE0, cmd=0x10
fido_tx: buf=000001A255DA4D60, len=319
0000: 02 a5 01 77 6c 62 73 64 6f 2e 64 6f 75 62 6c 65
0016: 6f 63 74 6f 70 75 73 2e 6e 65 74 02 58 20 a2 fe
0032: ee a3 57 46 c3 8f 60 3e 6e 0d 6b 75 30 d9 c9 43
0048: 88 22 5a 69 60 1a 19 9d eb 0a 05 59 4b 36 03 81
0064: a2 62 69 64 58 50 e7 23 e6 fe df d4 ef af cf 11
0080: 43 11 6c 03 dd a4 0f 50 37 13 0e 58 f6 09 07 48
0096: 29 9c 0b 61 fb 69 ac da a4 e2 3f d1 fc 92 06 0c
0112: 76 99 87 94 b3 80 f4 7f 6f 36 2e 6c 68 6e cd f5
0128: 04 c0 80 2c e7 f7 00 14 d5 97 c6 cf 15 02 fb df
0144: 96 16 3a ab fa 75 64 74 79 70 65 6a 70 75 62 6c
0160: 69 63 2d 6b 65 79 04 a1 6b 68 6d 61 63 2d 73 65
0176: 63 72 65 74 a3 01 a5 01 02 03 38 18 20 01 21 58
0192: 20 ef 9f d4 2b 72 5f 57 01 f4 61 36 4e 91 7d c4
0208: f9 da 58 2f 5f 37 cd d0 42 ba f4 32 b2 6f 43 bb
0224: 5d 22 58 20 9c f2 9b 80 70 83 d1 23 f4 75 1c 0f
0240: 32 66 73 d0 1f 15 eb ab 8c 99 f7 21 53 4d 48 bc
0256: db 11 0a d1 02 58 20 2c d5 83 f5 d1 47 e9 81 f0
0272: 80 ec 02 b4 1e 7b 6c 61 a0 6d 4e b7 46 7d 38 8d
0288: c7 93 50 5c ca 9c e8 03 50 d0 37 27 8e 98 66 b0
0304: b6 e1 f8 2f 9c af 2a c3 34 05 a1 62 75 70 f4
fido_pcsc_write: writing: buf=000000700BAFF010, len=246
0000: 90 10 00 00 f0 02 a5 01 77 6c 62 73 64 6f 2e 64
0016: 6f 75 62 6c 65 6f 63 74 6f 70 75 73 2e 6e 65 74
0032: 02 58 20 a2 fe ee a3 57 46 c3 8f 60 3e 6e 0d 6b
0048: 75 30 d9 c9 43 88 22 5a 69 60 1a 19 9d eb 0a 05
0064: 59 4b 36 03 81 a2 62 69 64 58 50 e7 23 e6 fe df
0080: d4 ef af cf 11 43 11 6c 03 dd a4 0f 50 37 13 0e
0096: 58 f6 09 07 48 29 9c 0b 61 fb 69 ac da a4 e2 3f
0112: d1 fc 92 06 0c 76 99 87 94 b3 80 f4 7f 6f 36 2e
0128: 6c 68 6e cd f5 04 c0 80 2c e7 f7 00 14 d5 97 c6
0144: cf 15 02 fb df 96 16 3a ab fa 75 64 74 79 70 65
0160: 6a 70 75 62 6c 69 63 2d 6b 65 79 04 a1 6b 68 6d
0176: 61 63 2d 73 65 63 72 65 74 a3 01 a5 01 02 03 38
0192: 18 20 01 21 58 20 ef 9f d4 2b 72 5f 57 01 f4 61
0208: 36 4e 91 7d c4 f9 da 58 2f 5f 37 cd d0 42 ba f4
0224: 32 b2 6f 43 bb 5d 22 58 20 9c f2 9b 80 70 83 d1
0240: 23 f4 75 1c 0f 00
fido_pcsc_write: read: buf=000001A255D8D668, len=2
0000: 90 00
fido_pcsc_read: reading: buf=000001A255D8D668, len=2
0000: 90 00
fido_pcsc_write: writing: buf=000000700BAFF010, len=85
0000: 80 10 00 00 4f 32 66 73 d0 1f 15 eb ab 8c 99 f7
0016: 21 53 4d 48 bc db 11 0a d1 02 58 20 2c d5 83 f5
0032: d1 47 e9 81 f0 80 ec 02 b4 1e 7b 6c 61 a0 6d 4e
0048: b7 46 7d 38 8d c7 93 50 5c ca 9c e8 03 50 d0 37
0064: 27 8e 98 66 b0 b6 e1 f8 2f 9c af 2a c3 34 05 a1
0080: 62 75 70 f4 00
fido_pcsc_write: read: buf=000001A255D8D668, len=3
0000: e1 90 00
fido_rx: dev=000001A255D6CFE0, cmd=0x10, ms=-1
fido_pcsc_read: reading: buf=000001A255D8D668, len=3
0000: e1 90 00
cbor_parse_reply: blob[0]=0xe1
fido_dev_get_assert_rx: adjust_assert_count
fido_tx: dev=000001A255D6CFE0, cmd=0x11
fido_tx: buf=0000000000000000, len=0
fido_nfc_tx: cmd=11
fido_dev_get_assert: FIDO_ERR_UNKNOWN (0xe1)

**Comment by kongeo** [comment]
Hello, could you please describe in more detail the registration and authentication flows as well as the overall use case (RP, authenticator etc)?
